flutter/Graminsta/lib/main.dart

```diff
@@ -59,71 +58,75 @@ class _HomePageState extends State<HomePage> {
     }
   }
 
-  getUserToken() async {
+  Future<bool> _checkUserToken() async {
     final prefs = await SharedPreferences.getInstance();
     final token = prefs.getString('user_token') ?? '';
-    debugPrint('user_token: $token');
-    return token;
+    if (token == "") {
+      Navigator.of(context).pushNamedAndRemoveUntil(
+        '/auth',
+        (route) => route == null,
+      );
+      return false;
+    }
+    return true;
```

Comment:
We might need to call backend to check whether the token has expired or not. But you can do this in another PR.